crates/node/src/scheduler/mod.rs

```diff
@@ -423,13 +422,84 @@ impl TaskManager for Scheduler {
                 );
             }
 
+            //move and build output files of the Tx execution
+            let executed_files: Vec<(File<Vm>, File<ProofVerif>)> = result
+                .files
+                .into_iter()
+                .map(|file| {
+                    let vm_file = File::<Vm>::new(
+                        file.path.to_string(),
+                        file.checksum[..].into(),
+                        running_task.task.tx,
+                    );
+
+                    let uuid = Uuid::new_v4();
+                    //format file_name to keep the current filename and the uuid.
+                    //put uuid at the end so that the uuid is use to save the file.
+                    let new_file_name = format!("{}/{}", file.path, uuid);
+                    let dest = File::<ProofVerif>::new(
+                        file.path,
+                        self.http_download_host.clone(),
+                        file.checksum[..].into(),
+                    );
+                    (vm_file, dest)
```

Comment:
I use the uuid because from my understanding all nodes execute the same program that generate the same file and all the generated files are downloaded on all nodes, so I need a way to have a unique name for the same file content (same hash).  I only use the uuid and not a more complex file path because I suppose that files are not directly accessed by the user but only with a RPC request that map the content with the name. So all file data are stored in the db and the filename is only an uuid.

Reply:
> from my understanding all nodes execute the same program that generate the same file and all the generated files are downloaded on all nodes, so I need a way to have a unique name for the same file content (same hash).

But why the file name has to be unique if the content is same? What use case requires that there are copies of same file for each node's execution, if the file content is bit-by-bit same?

> So all file data are stored in the db and the filename is only an uuid.

File data or metadata? Please, put the file contents on disk in file and only use DB for possibly helpful metadata :slightly_smiling_face: 

This makes debugging, backups and tool development a bit easier :slightly_smiling_face:

Reply:
Ok, I was thinking that we need to keep all node generated data for verification / history reason. In my understanding, 2 same files generated on 2 distinct nodes wasn't the same because we need to be able to prove or give all the data of both node.
So I change to the checksum. I put the file in {txhash}/{checksum}/{filename}